src/bp/core/repositories/events.ts

```diff
@@ -0,0 +1,64 @@
+import * as sdk from 'botpress/sdk'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+
+import Database from '../database'
+import { TYPES } from '../types'
+
+export interface EventRepository {
+  findEvents(fields: Partial<sdk.IO.StoredEvent>, searchParams: sdk.EventSearchParams)
+  deleteBeforeDate(date: Date): Promise<void>
+}
+
+export const DefaultSearchParams: sdk.EventSearchParams = {
+  sortOrder: [{ column: 'createdOn' }],
+  from: 0,
+  count: 10
+}
+
+const UNLIMITED_ELEMENTS = -1
+
+@injectable()
+export class KnexEventRepository implements EventRepository {
```

Comment:
is there another impl than `KnexEventRepository` ? if no then why define an interface for this ?

Reply:
I kept the same convention as all other repositories. But yes it could be simplified